tests/simulator/per_test.h

```diff
@@ -0,0 +1,144 @@
+#ifndef PHARE_TEST_SIMULATOR_PER_TEST_H
+#define PHARE_TEST_SIMULATOR_PER_TEST_H
+
+#include "simulator/simulator.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+// Having multiple PythonDataProvider per binary execution doesn't work so well
+
+struct __attribute__((visibility("hidden"))) StaticIntepreter
+{
+    std::shared_ptr<PHARE::initializer::PythonDataProvider> input;
+
+    StaticIntepreter()
+        : input{std::make_shared<PHARE::initializer::PythonDataProvider>()}
+    {
+        input->read();
+    }
+    static void set(PHARE::initializer::PHAREDict& _dict)
+    {
+        auto& dict = PHARE::initializer::PHAREDictHandler::INSTANCE().dict();
+        // hax - migrate to pure python *eventually*
+        if (dict.contains("simulation"))
+        {
+            dict["simulation"]["ions"]       = _dict["ions"];
```

Comment:
if this is not used then maybe remove it?

Reply:
done